core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -50,6 +51,21 @@ class SparkException(
   def this(message: String) =
     this(message = message, cause = null)
 
+  def this(
+      errorClass: String,
+      messageParameters: Array[String],
+      cause: Throwable,
+      context: Array[QueryContext],
+      summary: String) =
+    this(
+      message = SparkThrowableHelper.getMessage(errorClass, null, messageParameters, summary),
+      cause = cause,
+      errorClass = Some(errorClass),
+      errorSubClass = None,
+      messageParameters = messageParameters,
+      context
+    )
```

Comment:
nit: see the code around
```suggestion
      context)
```